v0/util/index.js

```diff
@@ -488,6 +495,42 @@ const handleMetadataForValue = (value, metadata) => {
       );
     }
   }
+
+  // handle multikeyMap
+  // sourceVal is expected to be an array
+  // if value is present in sourceVal, returns the destVal
+  // else returns the original value
+  // Expected multikeyMap value:
+  // "multikeyMap": [
+  //   {
+  //     "sourceVal": ["m", "M", "Male", "male"],
+  //     "destVal": "M"
+  //   },
+  //   {
+  //     "sourceVal": ["f", "F", "Female", "female"],
+  //     "destVal": "F"
+  //   }
+  // ]
+  if (multikeyMap) {
+    if (Array.isArray(multikeyMap)) {
+      multikeyMap.some(map => {
+        if (!map.sourceVal || !map.destVal || !Array.isArray(map.sourceVal)) {
+          logger.warn(
+            "multikeyMap skipped: sourceVal and destVal must be of valid type"
+          );
+          return true;
+        }
+
+        if (map.sourceVal.includes(formattedVal)) {
```

Comment:
can we 
 var upperCaseSourceValMap=  map.sourceVal.map(a => {
          return a.toUpperCase();
        });
  
        if (upperCaseSourceValMap.includes(formattedVal.toUpperCase())) {
          formattedVal = map.destVal;
          return true;
        }

So that even if "FEMALE" comes it is handled. and from sourceVal we can only set "m" "f" and "M" "F" is not required? WDYT?

Reply:
Yes, we can do this we can reduce case-specific keys. Either we can convert all keys to upper or lower-case and then compare the formattedVal in that case, we should mention this in the comment docs for ease of understanding Even though code is self-explanatory. We should do this @arajguha

Reply:
W.r.t I am getting a few things in mind. 
`
var upperCaseSourceValMap= map.sourceVal.map(a => {
return a.toUpperCase();
});
`
This map function to convert to upper or lower case would iterate through **all** the values and return the new array and on that new array we are looking for the match. So, in terms of iteration , wouldn't just one linear search be better ?

Second thing is, in this particular case, it happens to be an array of strings. For non-string values, this would crash. Would it better to generalise it?
What do you think ?

Reply:
I agree, for other `data type` and considering the time to iterate and create a new map of `uppercase values`. We should just accept an array of keys. It should be up to the integration developer how he wants to match. @ruchiramoitra  WDYT seems like a valid point?

Reply:
yes.. let's leave this. in the data json file add "FEMALE" and "MALE"

Reply:
okay. got it.